proxy/logging/LogFilter.h

```diff
@@ -394,44 +458,17 @@ wipeField(char **field, char *pattern, const char *uppercase_field)
   const char *lookup_dest = uppercase_field ? uppercase_field : *field;
 
   if (buf_dest) {
-    char *query_param              = strstr(buf_dest, "?");
-    const char *lookup_query_param = strstr(lookup_dest, "?");
+    char *query_param              = strchr(buf_dest, '?');
+    const char *lookup_query_param = strchr(lookup_dest, '?');
     if (!query_param || !lookup_query_param) {
       return;
     }
 
-    const char *p1 = strstr(lookup_query_param, pattern);
-    int field_pos  = p1 - lookup_query_param;
-    p1             = query_param + field_pos;
-
-    if (p1) {
-      char tmp_text[strlen(buf_dest) + 10];
-      char *temp_text = tmp_text;
-      memcpy(temp_text, buf_dest, (p1 - buf_dest));
-      temp_text += (p1 - buf_dest);
-      const char *p2 = strstr(p1, "=");
-      if (p2) {
-        p2++;
-        memcpy(temp_text, p1, (p2 - p1));
-        temp_text += (p2 - p1);
-        const char *p3 = strstr(p2, "&");
-        if (p3) {
-          for (int i = 0; i < (p3 - p2); i++) {
-            temp_text[i] = 'X';
-          }
-          temp_text += (p3 - p2);
-          memcpy(temp_text, p3, ((buf_dest + strlen(buf_dest)) - p3));
-        } else {
-          for (int i = 0; i < ((buf_dest + strlen(buf_dest)) - p2); i++) {
-            temp_text[i] = 'X';
-          }
-        }
-      } else {
-        return;
-      }
-
-      tmp_text[strlen(buf_dest)] = '\0';
-      strcpy(*field, tmp_text);
+    const char *pattern_in_param_name = findPatternFromParamName(lookup_query_param, pattern);
+    if (pattern_in_param_name) {
+      int field_pos         = pattern_in_param_name - lookup_query_param;
+      pattern_in_param_name = query_param + field_pos;
+      updatePatternForFieldValue(field, pattern_in_param_name, field_pos, buf_dest);
```

Comment:
The code doesn't seem to be capable to handle the case `http:///path?email=aaa&email=bbb&email=another_email`?